libutils/sequence.c

```diff
@@ -107,6 +107,13 @@ void SeqSet(Seq *seq, size_t index, void *item)
     seq->data[index] = item;
 }
 
+void SeqSoftSet(Seq *seq, size_t index, void *item)
```

Comment:
Just a wild idea. Would it make sense to actually return the original item? That would allow us to skip using `SeqAt` before setting.
This (like in the [test](https://github.com/cfengine/libntech/pull/147/files#diff-065eb547f1838448271cae05ff88bf1111fde1e9d11708f574a93a9a841b5be1R61-R65))
```c
item = SeqAt(seq, 5);
SeqSoftSet(seq, 5, new_item);
```
could become
```c
item = SeqSoftSet(seq, 5)
```

Reply:
> Just a wild idea. Would it make sense to actually return the original item? That would allow us to skip using `SeqAt` before setting.

That could be added as another function like `SeqGetAndSet()`. This is just in the idea of other `Soft` functions that we have.